### PR TITLE
 Automatically scrolls to current day in timetable

### LINKED ISSUE #33: Automatically scroll to current day in timetable

### PR DESCRIPTION
**Pull request by asdfMaciej**
Fixes #33

### CODE REVIEW

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 1]
```diff
@@ -54,15 +55,23 @@ class TimetableFragment : LifecycleFragment() {
                 fragment_timetable_message.visibility = View.GONE
 
                 val items = mutableListOf<IFlexible<*>>()
+                val days_n = mutableListOf<Int>()
```
Comment (shymmq) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: Create current date here as LocalDate.now()

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 2]
```diff
@@ -54,15 +55,23 @@ class TimetableFragment : LifecycleFragment() {
...
+                var current_n = 0
 
                 for ((date, schoolDay) in timetableData) {
+                    days_n.add(current_n)
```
Comment (shymmq) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: Compare date to today's date if they equal, save the header element

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 3]
```diff
@@ -80,9 +89,15 @@ class TimetableFragment : LifecycleFragment() {
...
 
+                val calendar = Calendar.getInstance()
+                val _day = calendar.get(Calendar.DAY_OF_WEEK)
+                val day = if (_day == 1) (6) else (_day - 2)  // Calendar counts sunday as first day
+
                 fragment_timetable_recyclerview.layoutManager = LinearLayoutManager(activity)
                 fragment_timetable_recyclerview.itemAnimator = null
                 fragment_timetable_recyclerview.adapter = adapter
+
+                adapter!!.recyclerView!!.scrollToPosition(days_n.get(day))
```
Comment (shymmq) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: scroll to todays header element if exists instead of index

**Review by shymmq — CHANGES_REQUESTED** [review]

**Comment by asdfMaciej** [comment]
Alright, will fix in spare time (probably tomorrow)

**Comment by asdfMaciej** [comment]
I've addressed to the changes

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 4]
```diff
@@ -54,20 +55,24 @@ class TimetableFragment : LifecycleFragment() {
...
-                val days_n = mutableListOf<Int>()
-                var current_n = 0
+                val today_day = Calendar.getInstance().get(Calendar.DAY_OF_WEEK)
+                var today_header = 0
 
                 for ((date, schoolDay) in timetableData) {
-                    days_n.add(current_n)
                     val header = LessonHeaderItem(date)
-                    current_n += 1
+                    val c = Calendar.getInstance()
```
Comment (rowysock) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: Please don't use old Java calendars. Use Joda instead. You got already prepared LocalDate date here

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 5]
```diff
@@ -54,20 +55,24 @@ class TimetableFragment : LifecycleFragment() {
                 fragment_timetable_recyclerview.visibility = View.VISIBLE
                 fragment_timetable_message.visibility = View.GONE
 
+                var _n = 0  // increases with every element
                 val items = mutableListOf<IFlexible<*>>()
-                val days_n = mutableListOf<Int>()
-                var current_n = 0
+                val today_day = Calendar.getInstance().get(Calendar.DAY_OF_WEEK)
```
Comment (rowysock) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: Make it Joda LocalDate
Also in Java camelCase is preffered over underscore_case

File: app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt [hunk 6]
```diff
@@ -54,20 +55,24 @@ class TimetableFragment : LifecycleFragment() {
                 fragment_timetable_recyclerview.visibility = View.VISIBLE
                 fragment_timetable_message.visibility = View.GONE
 
+                var _n = 0  // increases with every element
```
Comment (rowysock) [inline app/src/main/java/com/wabadaba/dziennik/ui/timetable/TimetableFragment.kt]: No need for this counter. scrollToPosition accepts both indices and elements

**Review by rowysock — CHANGES_REQUESTED** [review]

**Pull request closed without merging**